File: Code/C_Code/27.1.1_TouchSensor/TouchSensor.c
/**********************************************************************
* Filename    : TouchSensor.c
* Description : Touch Sensor TTP223 Control LED brightness. 
* Author      : www.freenove.com
* modification: 2022/4/20
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#include <softPwm.h>
#define ledPin    0  	    //define the ledPin
#define SensorPin 1      //define the SensorPin

int SensorState=LOW;    //store the State of button
int lastSensorState=LOW;//store the lastState of Sensor
long lastChangeTime;    //store the change time of Sensor state
long captureTime=10;   //set the stable time for Sensor state 
int Sensorvalue;
int grade=0;

int main(void)
{
    printf("Program is starting...\n");
    wiringPiSetup();              //Initialize wiringPi.	
    softPwmCreate(ledPin, 0, 100);//Creat SoftPWM pin
    pinMode(SensorPin, INPUT);    //Set SensorPin to input
    pullUpDnControl(SensorPin, PUD_DOWN);   //pull up to high level
    while(1){
         Sensorvalue = digitalRead(SensorPin);   //read the current state of Sensor
         if( Sensorvalue != lastSensorState){    //if the Sensor state has changed, record the time point
              lastChangeTime = millis();
        }
        //if changing-state of the Sensor last beyond the time we set, we consider that
         //the current Sensor state is an effective change rather than a buffeting
         if(millis() - lastChangeTime > captureTime){
             //if sensor state is changed, update the data.
              if(Sensorvalue != SensorState){
                  SensorState = Sensorvalue;
                  //if the state is low, it means the action is pressing
                  if(SensorState == HIGH){
                      grade=grade+1;
                      printf("Sensor is pressed!\n");
                  }
              }
         }
         else if(grade==1){
             softPwmWrite(ledPin, 35);
         }
         else if(grade==2){
             softPwmWrite(ledPin, 65);
         }
         else if(grade==3){
             softPwmWrite(ledPin, 100);
         }
         else if(grade==4){
             grade=0;
         }
         else {
             softPwmWrite(ledPin, 0);
         }
         lastSensorState = Sensorvalue;
    }
    return 0;
}

File: Code/Python_GPIOZero_Code/27.1.1_TouchSensor/TouchSensor.py
#!/usr/bin/env python3
#############################################################################
# Filename    : TouchSensor.py
# Description : Touch Sensor TTP223 Control LED brightness.
# Author      : www.freenove.com
# modification: 2023/05/13
#####################PW###################################################
from gpiozero import PWMLED
from sensor import TouchSensor
import time

ledPin    = 17     # define ledPin
sensorPin = 18     # define sensorPin
led    = PWMLED(ledPin)    
sensor = TouchSensor(sensorPin, pull_up=False)
grade=0

# Define the functions that will be called when the line is
def SensorEvent():
    global  grade
    grade=grade+1
    print("Sensor is pressed!")
    if(grade > 3):
        grade=0

def loop():
    sensor.when_touch = SensorEvent
    while True:
        global grade
        if grade==1:
            dc=35
            led.value = dc / 100.0     # set dc value as the duty cycle
        elif grade==2:
            dc=65
            led.value = dc / 100.0     # set dc value as the duty cycle
        elif grade==3:
            dc=100
            led.value = dc / 100.0     # set dc value as the duty cycle
        else :
            dc=0
            led.value = dc / 100.0     # set dc value as the duty cycle

def destroy():
    led.close() 
    sensor.close()                     

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()
        print("Ending program")